dialogue-apache-hc4-client/src/main/java/com/palantir/dialogue/hc4/ApacheHttpClientChannels.java

```diff
@@ -75,15 +75,26 @@
 
     private ApacheHttpClientChannels() {}
 
-    public static Channel create(ClientConfiguration conf, UserAgent baseAgent) {
+    public static Channel create(ClientConfiguration conf, UserAgent agent) {
+        CloseableHttpClient client = createCloseableHttpClient(conf);
+        List<Channel> channels =
+                conf.uris().stream().map(uri -> createSingleUri(uri, client)).collect(Collectors.toList());
+
+        return Channels.create(channels, agent, conf);
+    }
+
+    public static Channel createSingleUri(String uri, CloseableHttpClient client) {
+        return BlockingChannelAdapter.of(new ApacheHttpClientBlockingChannel(client, url(uri)));
```

Comment:
Another possibility is that we create a new client for each URI for additional isolation and avoiding contention in the connection pool.

Reply:
This API gives us the freedom to do that from our WC facade, because we can choose to pass in the same closeablehttpclient or not.